server/base/src/main/java/org/apache/accumulo/server/metadata/iterators/SetEqualityIterator.java

```diff
@@ -82,6 +86,13 @@ public void seek(Range range, Collection<ByteSequence> columnFamilies, boolean i
         byte[] ba = source.getTopKey().getColumnQualifierData().toArray();
         dos.writeInt(ba.length);
         dos.write(ba, 0, ba.length);
+
+        if (includeValue) {
+          byte[] valueData = source.getTopValue().get();
+          dos.writeInt(valueData.length);
+          dos.write(valueData, 0, valueData.length);
+        }
+
```

Comment:
This needs be structured a bit differently to encode the map on the server side in the same way its being encoded on the client side.

```suggestion
      if (includeValue) {
           //encode the key+value same way as the client side.  So encode both into a single byte array and then write that byte array out to 'dos'
       
           
      } else {
      // only encode the key/qualifier
        byte[] ba = source.getTopKey().getColumnQualifierData().toArray();
        dos.writeInt(ba.length);
        dos.write(ba, 0, ba.length);
      }
```

Reply:
Implemented the recommended modifications and pushed the changes. Thank you for your review.